Assets/Code/Scripts/PlayerMovementController.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        private void VelocityUpdate()
+        {
+            // Forward/Backward update
+            if (_moveForward && _velocityZ < maxVelocity)
+            {
+                _velocityZ += Time.deltaTime * acceleration;
+            } 
+            else if (!_moveForward && _velocityZ > 0.0f)
+            {
+                _velocityZ -= Time.deltaTime * decelerationFactor  * acceleration;
+            }
+            if (_moveBackward && _velocityZ > -maxVelocity)
+            {
+                _velocityZ -= Time.deltaTime * acceleration;
+            } 
+            else if (!_moveBackward  && _velocityZ < 0.0f)
+            {
+                _velocityZ += Time.deltaTime * decelerationFactor  * acceleration;
+            }
+            // Left/Right update
+            if (_moveLeft && _velocityX > -maxVelocity)
+            {
+                _velocityX -= Time.deltaTime * acceleration;
+            }
+            else if (!_moveLeft && _velocityX < 0.0f)
+            {
+                _velocityX += Time.deltaTime * decelerationFactor  * acceleration;
+            }
+            if (_moveRight && _velocityX < maxVelocity)
+            {
+                _velocityX += Time.deltaTime * acceleration;
+            }
+            else if (!_moveRight && _velocityX > 0.0f)
+            {
+                _velocityX -= Time.deltaTime * decelerationFactor  * acceleration;
+            }
+        }
+
+        private void ConstrainVelocity()
+        {
+            //Forward/Backward constraints
+            if (_moveForward && _velocityZ > maxVelocity)
+            {
+                _velocityZ = maxVelocity;
+            }
+            if (_moveBackward && _velocityZ < -maxVelocity)
+            {
+                _velocityZ = -maxVelocity;
+            }
+            //Left/Right constraints
+            if (_moveLeft && _velocityX < -maxVelocity)
+            {
+                _velocityX = -maxVelocity;
+            }
+            if (_moveRight && _velocityX > maxVelocity)
+            {
+                _velocityX = maxVelocity;
+            }
+        }
+        
+        void Update()
+        {
+            _moveForward = Input.GetKey(MoveForwardKey);
+            _moveLeft = Input.GetKey(MoveLeftKey);
+            _moveRight = Input.GetKey(MoveRightKey);
+            _moveBackward = Input.GetKey(MoveBackwardKey);
+            
+            VelocityUpdate();
+            ConstrainVelocity();
+            
+            _animator.SetFloat(VelocityXHash, _velocityX);
+            _animator.SetFloat(VelocityZHash, _velocityZ);
+            
+            BoundCheck();
+        }
+
+        // Places the player on the opposite side of the plane if they try to leave it.
+        private void BoundCheck()
+        {
+            var newPos = new Vector3();
+            for (var i = 0; i < 3; i+=2)
+            {
+                if (transform.position[i] < _planePosition[i]-_planeDimensions[i]/2.0f)
```

Comment:
just to make sure: you want to do `_planePosition[i] - (-planeDimensions[i]/2)`?